arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/stream/TerminalOps.kt

```diff
@@ -58,9 +58,22 @@ suspend fun <O> Stream<O>.firstOrNull(): O? =
  *
  * This a terminal operator, meaning this functions `suspend`s until the [Stream] finishes.
  * If any errors are raised while streaming, it's thrown from this `suspend` scope.
+ *
+ * It's possible to include an [alternative] function which can be used to return an alternative value or fail with an exception.
+ *
+ * ```kotlin:ank:playground
+ * import arrow.fx.coroutines.stream.*
+ *
+ * //sampleStart
+ * suspend fun main(): Unit =
+ *   Stream.empty<Int>()
+ *     .firstOrElse { error("Oops!") } // equivalent to .firstOrElse { throw IllegalStateException("Oops!") }
+ *     .let(::println) // not reached, IllegalStateException("Oops!") is thrown
+ * //sampleEnd
+ * ```
  */
-suspend fun <O> Stream<O>.firstOrError(): O =
-  firstOrNull() ?: throw NoSuchElementException()
+suspend fun <O> Stream<O>.firstOrElse(alternative: suspend () -> O): O =
```

Comment:
Is there a difference between having `alternative` be `suspend` and the whole function being `inline` instead?